tests/unit/plugins/modules/test_snapshot.py

```diff
@@ -0,0 +1,481 @@
+# Copyright: (c) 2024, Dell Technologies
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""Unit Tests for Snapshot module on PowerMax"""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import pytest
+from mock.mock import MagicMock
+from ansible_collections.dellemc.powermax.plugins.module_utils.storage.dell \
+    import utils
+
+utils.get_logger = MagicMock()
+utils.has_pyu4v_sdk = MagicMock(return_value=True)
+utils.pyu4v_version_check = MagicMock(return_value=None)
+utils.universion_check = MagicMock(return_value={"is_valid_universion": True})
+utils.get_U4V_connection = MagicMock()
+utils.close_connection = MagicMock()
+from ansible.module_utils import basic
+basic.AnsibleModule = MagicMock()
+
+
+from ansible_collections.dellemc.powermax.plugins.modules.snapshot import Snapshot, SnapshotHandler
+from ansible_collections.dellemc.powermax.tests.unit.plugins.module_utils.mock_snapshot_api import MockSnapshotApi
+from ansible_collections.dellemc.powermax.tests.unit.plugins.module_utils.mock_api_exception \
+    import MockApiException
+from ansible_collections.dellemc.powermax.tests.unit.plugins.module_utils.shared_library.fail_json \
+    import FailJsonException
+
+
+class TestSnapshot():
+    MODULE_PATH = 'ansible_collections.dellemc.powermax.plugins.modules.snapshot.Snapshot.'
+    MODULE_UTILS_PATH = 'ansible_collections.dellemc.powermax.plugins.module_utils.storage.dell.utils'
+    snapshot_args = MockSnapshotApi.SNAPSHOT_COMMON_ARGS
+
+    @pytest.fixture
+    def snapshot_module_mock(self, mocker):
+        snapshot_module_mock = Snapshot()
+        snapshot_module_mock.module = MagicMock()
+        snapshot_module_mock.replication = MagicMock()
+        snapshot_module_mock.module.check_mode = False
+        snapshot_module_mock.u4v_conn = MagicMock()
+        return snapshot_module_mock
+
+    def capture_fail_json_call(self, error_msg, snapshot_module_mock):
```

Comment:
Reuse this method: https://github.com/dell/ansible-powermax/pull/47/files#diff-b18360c64a5ae047d24b0e0c4a1ec2bd5ac9a5825c0f5a70ad30bfdbcbee0a13R30